api/grpcserver/events.go

```diff
@@ -0,0 +1,17 @@
+package grpcserver
+
+const subscriptionChanBufSize = 1 << 16
+
+func consumeEvents(inputCh <-chan interface{}) <-chan interface{} {
+	outputCh := make(chan interface{}, subscriptionChanBufSize)
+
+	go func(inputCh <-chan interface{}) {
+		for event := range inputCh {
+			outputCh <- event
```

Comment:
this need to exit the loop if send is blocking, otherwise it doesn't change anything. eventually when buffer is full publisher will block

Reply:
@dshulyak If send is blocking, should we drop the peer after exiting the loop? If what you suggested is implemented, a peer wouldn't receive new events anymore after exiting the loop

Reply:
yes the stream must be terminated, preferably with an error that will explain that the subscriber was too slow (let's not use peer outside of the p2p context)